<?xml version="1.0"?>
<launch>

    <!--- Sim Time -->
    <param name="/use_sim_time" value="true" />
    <param name="scan_period" value="0.2" />
    <param name="scan_line" value="16" />
    <param name="vertical_angle" type="double" value="2.0" />
    <param name="max_dis" type="double" value="50.0" />
    <param name="min_dis" type="double" value="0.5" />
    <param name="sequence_number" type="int" value="05" />

    <node pkg="intensity_slam" type="intensity_slam_odom_estimation_node" name="intensity_slam_odom_estimation_node" output="screen"/>
    <node pkg="intensity_slam" type="intensity_slam_laser_processing_node" name="intensity_slam_laser_processing_node" output="screen"/>

    <node pkg="intensity_slam" type="intensity_slam_isc_generation_node" name="intensity_slam_isc_generation_node" output="screen"/>
    <node pkg="intensity_slam" type="intensity_slam_isc_optimization_node" name="intensity_slam_isc_optimization_node" output="screen"/>

    <node pkg="tf" type="static_transform_publisher" name="world2map_tf"  args="0 0 0 0 0 0 world map 100" />
    <arg name="rviz" default="true" />
    <group if="$(arg rviz)">
        <node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args="-d $(find intensity_slam)/rviz/intensity_slam.rviz" />
    </group>

  	<node pkg="hector_trajectory_server" type="hector_trajectory_server" name="trajectory_server_loam" ns="gt" >
        <param name="/target_frame_name" value="world" />
        <param name="/source_frame_name" value="DKCS2022" />
        <param name="/trajectory_update_rate" value="10.0" />
        <param name="/trajectory_publish_rate" value="10.0" />
    </node>

</launch>
